fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -235,36 +232,31 @@ class WebExtensionPromptFeature(
                 positiveButtonRadius =
                 (context.resources.getDimensionPixelSize(R.dimen.tab_corner_radius)).toFloat(),
             ),
-            onPositiveButtonClicked = {
-                onConfirm(true)
-                consumePromptRequest()
-            },
-            onNegativeButtonClicked = {
-                onConfirm(false)
-                consumePromptRequest()
-            },
+            onPositiveButtonClicked = { handlePermissions(promptRequest, granted = true) },
+            onNegativeButtonClicked = { handlePermissions(promptRequest, granted = false) },
         )
         dialog.show(
             fragmentManager,
             PERMISSIONS_DIALOG_FRAGMENT_TAG,
         )
     }
 
+    @Suppress("NestedBlockDepth")
```

Comment:
Nit: Maybe we don't need the suppression anymore as we are not adding more lines :)